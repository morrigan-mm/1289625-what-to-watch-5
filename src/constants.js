export const PageMovieTab = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const PageType = {
  MOVIE_CARD: `movie-card`,
  USER_PAGE: `user-page`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
