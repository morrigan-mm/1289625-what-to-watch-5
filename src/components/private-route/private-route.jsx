import React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {AppRoute, AuthorizationStatus} from "../../constants";
import {getUserStatus} from "../../store/selectors";

const PrivateRoute = (props) => {
  const {render, path, exact, authorizationStatus} = props;

  return (
    <Route
      path={path}
      exact={exact}
      render={(routeProps) => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH
            ? render(routeProps)
            : <Redirect to={AppRoute.LOGIN.url()} />
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  render: PropTypes.func.isRequired,
  path: PropTypes.string,
  exact: PropTypes.bool,
  authorizationStatus: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  authorizationStatus: getUserStatus(state)
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
