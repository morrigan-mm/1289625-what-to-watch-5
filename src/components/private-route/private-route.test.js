import React from "react";
import {StaticRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {AuthorizationStatus} from "../../constants";
import {PrivateRoute} from "./private-route";

describe(`PrivateRoute`, () => {
  it(`Should render with call given render function, if status is AUTH`, () => {
    const tree = renderer
      .create(
          <StaticRouter>
            <PrivateRoute
              authorizationStatus={AuthorizationStatus.AUTH}
              render={() => <div />}
            />
          </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render redirect, if status is NO_AUTH`, () => {
    const tree = renderer
      .create(
          <StaticRouter>
            <PrivateRoute
              authorizationStatus={AuthorizationStatus.NO_AUTH}
              render={() => <div />}
            />
          </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
