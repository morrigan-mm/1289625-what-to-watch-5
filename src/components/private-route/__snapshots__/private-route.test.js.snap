// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute Should render redirect, if status is NO_AUTH 1`] = `null`;

exports[`PrivateRoute Should render with call given render function, if status is AUTH 1`] = `<div />`;
